### ISSUE TITLE
`xgb.importance` fails

### ISSUE THREAD

**Issue by lukas-hager** [dmlc/xgboost#7540]
I've been using `xgb.importance` for a model, but after some feature tweaking, the function failed. The model file is enclosed -- I believe that this code snippet should reproduce the error:

```r
model = xgb.load(unzipped_model_file)
xgb.importance(model)
```

This code gives me the following error:

```
Error in `[.data.table`(td, isLeaf == FALSE, `:=`((branch_cols), { : 
  Supplied 9936 items to be assigned to 9937 items of column 'Feature'. If you wish to 'recycle' the RHS please use rep() to make this intent clear to readers of your code.
```
With the following traceback
```
4: `[.data.table`(td, isLeaf == FALSE, `:=`((branch_cols), {
       matches <- regmatches(t, regexec(branch_rx, t))
       xtr <- do.call(rbind, matches)[, c(2, 3, 5, 6, 7, 8, 10), 
           drop = FALSE]
       xtr[, 3:5] <- add.tree.id(xtr[, 3:5], Tree)
       as.data.table(xtr)
   }))
3: td[isLeaf == FALSE, `:=`((branch_cols), {
       matches <- regmatches(t, regexec(branch_rx, t))
       xtr <- do.call(rbind, matches)[, c(2, 3, 5, 6, 7, 8, 10), 
           drop = FALSE]
       xtr[, 3:5] <- add.tree.id(xtr[, 3:5], Tree)
       as.data.table(xtr)
   })]
2: xgb.model.dt.tree(feature_names = feature_names, text = model_text_dump, 
       trees = trees)
1: xgb.importance(model = model)
```
Any guidance you can provide would be extremely helpful! Apologies if this is obvious user error.
[season_xgb_no_adv.zip](https://github.com/dmlc/xgboost/files/7799936/season_xgb_no_adv.zip)

**Comment by trivialfis** [comment]
Hi, which xgboost version are you using?

**Comment by lukas-hager** [comment]
Hello! I'm using 1.3.2.1

**Comment by trivialfis** [comment]
I have reproduced it in 1.3 but I think it's fixed in 1.5.  Feel free to reopen if the problem persists on 1.5 for you.  Thanks for opening the issue.
